packages/@react-aria/test-utils/src/menu.ts

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {act, waitFor, within} from '@testing-library/react';
+import {triggerLongPress} from './events';
+import {UserOpts} from './user';
+
+interface MenuOptions extends UserOpts {
+  user: any
+}
+export class MenuTester {
+  private user;
+  private _interactionType: UserOpts['interactionType'];
+  private _advanceTimer: UserOpts['advanceTimer'];
+  private _trigger: HTMLElement;
+
+  constructor(opts: MenuOptions) {
+    this.user = opts.user;
+    this._interactionType = opts.interactionType || 'mouse';
+    this._advanceTimer = opts.advanceTimer;
+  }
+
+  setElement = (element: HTMLElement) => {
+    // Handle case where a submenu trigger is provided to the tester
+    if (element.getAttribute('role') === 'menuitem') {
+      this._trigger = element;
+    } else {
+      // Handle case where element provided is a wrapper of the trigger button
+      let trigger = within(element).queryByRole('button');
+      if (trigger) {
+        this._trigger = trigger;
+      }
+    }
+
+    if (this._trigger == null) {
+      this._trigger = element;
+    }
+  };
+
+  setInteractionType = (type: UserOpts['interactionType']) => {
+    this._interactionType = type;
+  };
+
+  // TODO: this has been common to select as well, maybe make select use it? Or make a generic method. Will need to make error messages generic
+  // One difference will be that it supports long press as well
+  open = async (opts: {needsLongPress?: boolean} = {}) => {
```

Comment:
add an option to suppress the alert? that way a user to can call open on a disabled menu and not have it throw?

Reply:
ah lol, just saw this after typing the reply in https://github.com/adobe/react-spectrum/pull/6208#discussion_r1739517638. I'll go ahead and add an option for "triggerDisabled" perhaps so it is more explicit

Reply:
actually I can just check for disabled like I do in combobox